testsuite/do_tests.sh.in

```diff
@@ -541,64 +466,20 @@ echo
 echo "Phase 8: Running C++ tests (experimental)"
 echo "-----------------------------------------"
 
-if command -v "${NEST_PATH}/run_all_cpptests" > /dev/null 2>&1; then
-  CPP_TEST_OUTPUT="$(${NEST_PATH}/run_all_cpptests 2>&1)"
-  # TODO:
-  # We should check the return code ($?) of run_all_cpptests here to see
-  # if a fatal crash occured. We cannot simply test $? != 0, since
-  # run_all_cpptests will return a non-zero code if tests fail.
-
-  # TODO:
-  # The regex for the total number seems fine according to
-  # https://www.boost.org/doc/libs/1_65_0/libs/test/doc/html/boost_test/test_output/log_formats/log_human_readable_format.html
-  # but does the check for failures work as it should?
-  # At some point, we should also count skipped tests.
-  CPP_TEST_TOTAL="$(echo "$CPP_TEST_OUTPUT" | sed -${EXTENDED_REGEX_PARAM} -n 's/.*Running ([0-9]+).*/\1/p')"
-  CPP_TEST_FAILED="$(echo "$CPP_TEST_OUTPUT" | sed -${EXTENDED_REGEX_PARAM} -n 's/.*([0-9]+) failure.*/\1/p')"
-
-  # replace empty strings by zero so arithmetic expressions below work
-  CPP_TEST_TOTAL=${CPP_TEST_TOTAL:-0}
-  CPP_TEST_FAILED=${CPP_TEST_FAILED:-0}
-  CPP_TEST_PASSED=$(( $CPP_TEST_TOTAL - $CPP_TEST_FAILED ))
-
-  TEST_TOTAL=$(( $TEST_TOTAL + $CPP_TEST_TOTAL ))
-  TEST_PASSED=$(( $TEST_PASSED + $CPP_TEST_PASSED ))
-  TEST_FAILED=$(( $TEST_FAILED + $CPP_TEST_FAILED ))
-
-  echo "  C++ tests : ${CPP_TEST_TOTAL}"
-  echo "     Passed : ${CPP_TEST_PASSED}"
-  echo "     Failed : ${CPP_TEST_FAILED}"
-
+if command -v run_all_cpptests > /dev/null 2>&1; then
+  CPP_TEST_OUTPUT=$( run_all_cpptests --logger=JUNIT,error,"${TEST_OUTDIR}/08_cpptests.xml":HRF,error,stdout 2>&1 )
+  echo "${CPP_TEST_OUTPUT}" | tail -2
 else
   echo "  Not running C++ tests because NEST was compiled without Boost."
 fi
 
-echo
-echo "NEST Testsuite Summary"
-echo "----------------------"
-echo "  NEST Executable: $(which nest)"
-echo "  SLI Executable : $(which sli)"
-echo "  Total number of tests: ${TEST_TOTAL}"
-echo "     Passed: ${TEST_PASSED}"
-echo "     Skipped: ${TEST_SKIPPED} ${PYNEST_TEST_SKIPPED_TEXT:-}"
-echo "     Failed: ${TEST_FAILED} ${PYNEST_TEST_FAILED_TEXT:-}"
-echo
 
-if test ${TEST_FAILED} -gt 0 ; then
-    echo "***"
-    echo "*** There were errors detected during the run of the NEST test suite!"
-    ask_results
-else
-    rm -rf "${TEST_TMPDIR}"
-fi
+@PYTHON@ "$(dirname $0)/summarize_tests.py" "${TEST_OUTDIR}"
```

Comment:
Probably you already know this, but anyway: I think the current failure in Travis is caused by `@PYTHON@` being empty in case we compiled without PyNEST, which leads to `summarize_tests.py` (incorrectly) being executed as a shell script due to a missing `#!/usr/bin/env python3` shebang in that file.

Reply:
Indeed. Interestingly, `parse_travis_log.py` also has no shebang and is invoked as `python extras/parse_travis_log.py travis_build.sh.log ;` from `travis.yml`.  What would be the most consistent and portable approach to running Python scripts during the build/test process?

Reply:
I think invoking as `python ...` is perfectly fine for running helper scripts as we don't require any specific Python version for them to do their job. `@PYTHON@` should only be used to run the actual Python tests, as this ensures that we actually use the exact version of Python that was used to compile against. Insofar, leaving out the shebang would be perfectly fine from my point of view.